buffer/src/main/java/io/netty/buffer/ByteBufUtil.java

```diff
@@ -706,6 +706,66 @@ public static void copy(AsciiString src, int srcIdx, ByteBuf dst, int length) {
     }
 
     /**
+     * Creates a retained duplicate of the specified {@link ByteBuf}.
+     */
+    public static ByteBuf rduplicate(ByteBuf buf) {
+        final int readerIndex = buf.readerIndex();
+        final int writerIndex = buf.writerIndex();
+
+        final AbstractByteBuf duplicate;
+        if (buf instanceof SlicedAbstractByteBuf) {
+            final SlicedAbstractByteBuf oldSlice = (SlicedAbstractByteBuf) buf;
+            duplicate = PooledSlicedByteBuf.newInstance(oldSlice.unwrap(), 0, buf.capacity(), oldSlice.adjustment());
```

Comment:
@ninja- - IIUC the meaning of `retainedDuplicate` (or `rduplicate`) is logically `duplicate().retain()`. Can you clarify how this implies pooling is involved? Pooling (or not) currently has a clean division from the `ByteBuf` API and is controlled by the allocator, I'm not sure why we would break that pattern for this new API.